.github/workflows/lint_and_test.yaml

```diff
@@ -41,10 +49,32 @@ jobs:
           restore-keys: |
             ${{ env.cache_name }}-${{ runner.os }}-go-
         env:
-          cache_name: run-tests-go-cache
+          cache_name: run-tests-go-cache # needs to be the same key in the end as in the build step
 
+      - name: Build
+        run: make build -j
       - name: Test
-        run: make build install-requirements test
+        run: make install-requirements test
```

Comment:
just from reading perspective this looks a bit odd... why are the requirements installed AFTER the build? Are those requirements only required for testing?

Reply:
that is a good question. Basically from what I understand these requirements are only for testing (e.g. the vault CLI), but not for building the binary

Reply:
should we then rename that make target to: 'install-test-requirements'?

Reply:
the requirements are unfortunately shared between other jobs as well so it will be hard to justify that. (e.g. make generate)

Reply:
I agree that its not that simple, since the "requirements" also include e.g. golang-ci-lint. Theoretically, I guess the best way would be to split the install-requirements target. And the test target would depend on a install-requirements. But that's probably out of scope for this PR.